app/controllers/admin/promotional_feature_items_controller.rb

```diff
@@ -56,15 +57,27 @@ def redirect_to_feature(notice = nil)
   end
 
   def promotional_feature_item_params
-    params.require(:promotional_feature_item).permit(
+    @promotional_feature_item_params ||= params.require(:promotional_feature_item).permit(
       :summary,
       :image,
       :image_alt_text,
       :title,
       :title_url,
       :double_width,
       :image_cache,
+      :youtube_video_url,
+      :image_or_youtube_video_url,
       links_attributes: %i[url text _destroy id],
     )
   end
+
+  def clean_image_or_youtube_video_url_param
+    case promotional_feature_item_params.delete(:image_or_youtube_video_url).to_s
```

Comment:
Oh rightI could've sworn it did when i was manually testing. Let me have a play. I can always delete the associated image post save if required.

Reply:
Sure, I was actually expecting that model.assign_attributes(image: nil) wouldn't actually reset the field since submitting the form would essentially have an empty image form input (since you don't set an image on most form submissions).

But does look like it does:

```
[6] pry(main)> pfi.image.present?
=> true
[7] pry(main)> pfi.assign_attributes(image: nil)
=> nil
[8] pry(main)> pfi.image.present?
=> false
```

I wasn't thinking of Asset Manager but that is indeed a great shout, you're right we'd leave an orphaned asset. Your code looks good, but I'm a bit worried about the use of `.youtube_video_url.present?` which makes me wonder if we also leave orphaned images already when someone updates the image. I'd imagine we'd want the check to be more if the image url has changed?

Reply:
Yea i think it wouldn't remove the image if we didn't mess with the params to manually set it to nil.

Great shout on change. I totally missed it'd still leave old images on change. I'll have a play now but i think something like

`delete_asset if legacy_url != current_url` 

would work

Reply:
I've pushed up some changes if you'd like a 👀

Reply:
Yup, that looks good. Nice one 👍